iac-extensions/spring-config/src/main/antlr/PropertiesParser.g4

```diff
@@ -1,11 +1,30 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
 parser grammar PropertiesParser;
 
 options {
     tokenVocab = PropertiesLexer;
 }
 
 propertiesFile
-    : LEADING_SPACING? row* EOF
+    : NEWLINE? (row NEWLINE?)* EOF
```

Comment:
It can be simplified to 
```suggestion
    : NEWLINE? row* EOF
```
as `NEWLINE` is a part of `eol` and each `row` ends with `eol`.